src/Marker.swift

```diff
@@ -0,0 +1,194 @@
+//
+//  Marker.swift
+//  ios-sdk
+//
+//  Created by Sarah Lensing on 3/22/17.
+//  Copyright © 2017 Mapzen. All rights reserved.
+//
+
+import Foundation
+import TangramMap
+
+public enum MarkerType : Int {
+  case currentLocation, searchPin, routeLine
+}
+
+@objc(MZMarker)
+public class Marker: NSObject {
+
+  private static let kPointStyle = "points"
+  private static let kLineStyle = "lines"
+  private static let kPolygonStyle = "polygons"
+  private static let kDefaultBackgroundColor = UIColor.white
+  private static let kDefaultInteractive = true
+  private static let kDefaultSize = CGSize.zero
```

Comment:
@msmollin In your case it's not possible that the stylesheet doesn't provide a size, since you use the overlay sprites referenced in the stylesheet.

The case you need to handle though is the user providing a `UIImage` but not size, in that case I would user the size of the `UIImage` as a potential default.